### ISSUE TITLE
v0.11.13 - panic: runtime error: index out of range 

### ISSUE THREAD

**Issue by jugg1es** [hashicorp/terraform#20712]
### Terraform Version

```
0.11.13
```

### Crash Output

```
goroutine 1 [running]:
github.com/hashicorp/terraform/config/module.Storage.loadManifest(0xc00001e2e0, 0x12, 0x33c4180, 0xc00036e340, 0x0, 0xc0003e9be0, 0xc0004309f0, 0x5b306a0, 0x0, 0x25, ...)
        /opt/gopath/src/github.com/hashicorp/terraform/config/module/storage.go:110 +0x487
github.com/hashicorp/terraform/config/module.Storage.moduleDir(0xc00001e2e0, 0x12, 0x33c4180, 0xc00036e340, 0x0, 0xc0003e9be0, 0xc000430a20, 0x27, 0x2, 0x2, ...)
        /opt/gopath/src/github.com/hashicorp/terraform/config/module/storage.go:185 +0x64
github.com/hashicorp/terraform/config/module.Storage.findModule(0xc00001e2e0, 0x12, 0x33c4180, 0xc00036e340, 0x0, 0xc0003e9be0, 0xc000430a20, 0x27, 0x0, 0x0, ...)
        /opt/gopath/src/github.com/hashicorp/terraform/config/module/storage.go:258 +0x91
github.com/hashicorp/terraform/config/module.(*Tree).getChildren(0xc000100680, 0xc00037e4e0, 0xc0001006b8, 0xb28719, 0x300d660)
        /opt/gopath/src/github.com/hashicorp/terraform/config/module/tree.go:234 +0x155e
github.com/hashicorp/terraform/config/module.(*Tree).Load(0xc000100680, 0xc00037e4e0, 0x0, 0x0)
        /opt/gopath/src/github.com/hashicorp/terraform/config/module/tree.go:178 +0xa5
github.com/hashicorp/terraform/command.(*Meta).Module(0xc000378680, 0xc00003ee60, 0x42, 0x0, 0x0, 0x0, 0x0)
        /opt/gopath/src/github.com/hashicorp/terraform/command/meta_new.go:57 +0xc2
github.com/hashicorp/terraform/command.(*PlanCommand).Run(0xc000378680, 0xc00036e380, 0x2, 0x2, 0xc0003030d0)
        /opt/gopath/src/github.com/hashicorp/terraform/command/plan.go:77 +0xaf1
github.com/mitchellh/cli.(*CLI).Run(0xc000334500, 0xc000334500, 0xc0001b5d20, 0x1)
        /opt/gopath/pkg/mod/github.com/mitchellh/cli@v0.0.0-20171129193617-33edc47170b5/cli.go:255 +0x20e
main.wrappedMain(0x0)
        /opt/gopath/src/github.com/hashicorp/terraform/main.go:223 +0xb55
main.realMain(0x0)
        /opt/gopath/src/github.com/hashicorp/terraform/main.go:100 +0xbe
main.main()
        /opt/gopath/src/github.com/hashicorp/terraform/main.go:36 +0x35
```

### Expected Behavior

Terraform commands like `plan` should work.

### Actual Behavior

Any attempt to run terraform throws the above error

### Steps to Reproduce

`terraform plan` or `terraform apply` or really any terraform command.

### Additional Context

Someone on our team just upgraded to 0.11.13 and everyone now has to upgrade but the executable crashes

**Comment by jugg1es** [comment]
Ok, I actually have an update.  We were upgrading from version 0.11.10.  If I delete the `.terraform` directory and run `init` again, it succeeds when I run terraform commands.  Seems like it's a backward-compatibility thing.  In any case, we found a way to resolve this.

**Comment by apparentlymart** [comment]
Thanks for reporting this, @jugg1es.

It seems like there was a slight change to how Terraform interprets the module metadata in the `.terraform` directory which has inadvertently introduced an incompatibility here. With 0.11.13 already out I imagine we can't do anything to mitigate it for that release, but perhaps we can track down what exactly has changed here and fix it for a hypothetical future v0.11.14 release.

(No such release is actually planned since we're hoping v0.12.0 will be the next stable one, but if others seem to be encountering this same crash then another release could be warranted to address it.)

**Comment by jakauppila** [comment]
I had encountered the same thing, but utilized the same workaround of just deleting the `.terraform` directory and running `init` again. I'm not dealing with any prod configs at this point, so not really an issue for us.

**Comment by jugg1es** [comment]
gonna close it since there is a workaround

**Comment by apparentlymart** [comment]
I'm going to reopen this for now since the team is still thinking about potential mitigation here, and also it's useful to have this issue open for now in case others find the same issue and can thus more easily find the workaround.

**Comment by bflad** [comment]
FYI another similar in nature report for this just came in: #20742

**Comment by mjiderhamn** [comment]
The error was introduced between 0.11.11 and 0.11.12 and remains in 0.11.13.

**Comment by apparentlymart** [comment]
Hi all!

This issue was fixed in Terraform 0.11.14. Sorry for the delay in fixing it; it got a bit held up by the preparation for the 0.12.0 release.
